plugins/inputs/smart/smart.go

```diff
@@ -18,27 +19,44 @@ import (
 )
 
 var (
-	execCommand = exec.Command // execCommand is used to mock commands in tests.
-
 	// Device Model:     APPLE SSD SM256E
-	modelInInfo = regexp.MustCompile("^Device Model:\\s+(.*)$")
+	modelInfo = regexp.MustCompile("^Device Model:\\s+(.*)$")
+	// Product:              HUH721212AL5204
+	productInfo = regexp.MustCompile("^Product:\\s+(.*)$")
+	// Model Number: TS128GMTE850
+	modelNoInfo = regexp.MustCompile("^Model Number:\\s+(.*)$")
 	// Serial Number:    S0X5NZBC422720
-	serialInInfo = regexp.MustCompile("^Serial Number:\\s+(.*)$")
+	serialInfo = regexp.MustCompile("^Serial Number:\\s+(.*)$")
 	// LU WWN Device Id: 5 002538 655584d30
-	wwnInInfo = regexp.MustCompile("^LU WWN Device Id:\\s+(.*)$")
+	wwnInfo = regexp.MustCompile("^LU WWN Device Id:\\s+(.*)$")
 	// User Capacity:    251,000,193,024 bytes [251 GB]
-	usercapacityInInfo = regexp.MustCompile("^User Capacity:\\s+([0-9,]+)\\s+bytes.*$")
+	usercapacityInfo = regexp.MustCompile("^User Capacity:\\s+([0-9,]+)\\s+bytes.*$")
 	// SMART support is: Enabled
-	smartEnabledInInfo = regexp.MustCompile("^SMART support is:\\s+(\\w+)$")
+	smartEnabledInfo = regexp.MustCompile("^SMART support is:\\s+(\\w+)$")
 	// SMART overall-health self-assessment test result: PASSED
 	// PASSED, FAILED, UNKNOWN
 	smartOverallHealth = regexp.MustCompile("^SMART overall-health self-assessment test result:\\s+(\\w+).*$")
+	// SMART Health Status: OK
+	smartHealth = regexp.MustCompile("^SMART Health Status:\\s+(\\w+).*$")
+
+	// Accumulated start-stop cycles:  7
+	sasStartStopAttr = regexp.MustCompile("^Accumulated start-stop cycles:\\s+(.*)$")
+	// Accumulated load-unload cycles:  39
+	sasLoadCycleAttr = regexp.MustCompile("^Accumulated load-unload cycles:\\s+(.*)$")
+	// Current Drive Temperature:     34 C
+	sasTempAttr = regexp.MustCompile("^Current Drive Temperature:\\s+(.*)\\s+C(.*)$")
+	// Temperature: 38 Celsius
+	nvmeTempAttr = regexp.MustCompile("^Temperature:\\s+(.*)\\s+(.*)$")
+	// Power Cycles: 472
+	nvmePowerCycleAttr = regexp.MustCompile("^Power Cycles:\\s+(.*)$")
+	// Power On Hours: 6,038
+	nvmePowerOnAttr = regexp.MustCompile("^Power On Hours:\\s+(.*)$")
```

Comment:
This is okay for now, but I think in the future we shouldn't add any additional fields until we have a more efficient way to parse the output.  We don't want to end up running 20+ regex per line.   Maybe we will classify the line using a single regex and then parse further once we know which pattern to use.

Reply:
Again, this isn't technically an additional field, just the way to parse it from a different device type.

Reply:
My concern is more about the way we are parsing the output, and how it will scale as we add additional regular expressions.  Right now it is essentially `number-of-lines * number-of-regex`, and for each field/attr we need another regex.  It should be possible to change this to `number-of-lines * 2` or better.

Reply:
I too found that disgusting. I had thought about refactoring that part, but felt the speed difference wouldn't be worth the effort. Maybe we should really evaluate that?

Reply:
It is probably fine for this pull request, but I don't think we can continue to add more regex with the current strategy for much longer.

Reply:
I don't see any Percentage Used included. I suppose this is one of the more important attributes to monitor. I know the regex way is not the way to go and the amount of attributes should be kept to a minimum, but could this very important attribute be added as well? Especially in high-write environments this is a vital measurement.

Reply:
One more won't be the end of the world, can you open a new feature request?

Reply:
@sg-jorrit I intentionally didn't include percentage used because that is already being collected (and seems to fit better) in the `disk` input plugin, but I suppose there's no harm in collecting the same thing in multiple places

Reply:
I think, but might be wrong, that the percentage used here is the SSD wear: how much of the expected write cycle of the drive has been used.

Reply:
Ah, now that makes sense